### ISSUE TITLE
Feature Request: Bumpwater using engine uniform buffer sun so it tracks sun position changes.

### ISSUE THREAD

**Issue by Beherith** [beyond-all-reason/spring#680]
Currently, bumpwater uses #define'd SUNDIR, instead of engine uniform buffers. For dynamic sun, it would be ideal if bumpwater shader wouldnt have to be recompiled for each change. 

Some other constants are:
fogColor
fogParams
GLSLDefineConstf1(definitions, "shadowDensity",  sunLighting->groundShadowDensity);

Thank you very much!

**Comment by lhog** [comment]
Wanna have dynamic sun position/color?

**Comment by Beherith** [comment]
Yeah, i have everything done on the lua side, only two blockers remain, bumpwater sun pos and ssmf not shading according to sun when underwater

**Comment by lhog** [comment]
So can you drop the full list of parameters you want to turn into uniforms?

**Comment by Beherith** [comment]
Thanks Ivand, there are only 3 parameters that are, to my knowledge already present in engine uniform buffers, that I would like connected from there, and also to prevent the recompilation of the bumpwater shader on Spring.Set***

https://github.com/beyond-all-reason/spring/blob/be954f434d069de16d5416ac14b98cd00d83b741/rts/Rendering/Env/BumpWater.cpp#L453
```GLSLDefineConstf3(definitions, "SunDir"       ,  ISky::GetSky()->GetLight()->GetLightDir()); // FIXME: not a constant```
Spring.SetSunDirection()

https://github.com/beyond-all-reason/spring/blob/be954f434d069de16d5416ac14b98cd00d83b741/rts/Rendering/Env/BumpWater.cpp#L469
```GLSLDefineConstf1(definitions, "shadowDensity",  sunLighting->groundShadowDensity);```
Spring.SetSunLighting

https://github.com/beyond-all-reason/spring/blob/be954f434d069de16d5416ac14b98cd00d83b741/cont/base/springcontent/shaders/GLSL/BumpWaterFS.glsl#L334
```  // FOG
    float fog = clamp( (gl_Fog.end - abs(gl_FogFragCoord)) * gl_Fog.scale ,0.0,1.0);
    gl_FragColor.rgb = mix(gl_Fog.color.rgb, gl_FragColor.rgb, fog );```
Spring.SetAtmosphere
```


There are a few others, of _much_ lower priority, that would be nice to have as uniforms, but these arent nearly as important:

https://github.com/beyond-all-reason/spring/blob/be954f434d069de16d5416ac14b98cd00d83b741/rts/Rendering/Env/BumpWater.cpp#L445-L448
``` 
	GLSLDefineConstf4(definitions, "SurfaceColor",   waterRendering->surfaceColor*0.4, waterRendering->surfaceAlpha );
	GLSLDefineConstf4(definitions, "PlaneColor",     waterRendering->planeColor*0.4, waterRendering->surfaceAlpha );
	GLSLDefineConstf3(definitions, "DiffuseColor",   waterRendering->diffuseColor);
	GLSLDefineConstf3(definitions, "SpecularColor",  waterRendering->specularColor);
```
